node/core/pvf/src/prepare/worker.rs

```diff
@@ -342,11 +357,26 @@ async fn recv_response(stream: &mut UnixStream, pid: u32) -> io::Result<PrepareR
 ///
 ///	7. Send the result of preparation back to the host. If any error occurred in the above steps, we
 ///	   send that in the `PrepareResult`.
-pub fn worker_entrypoint(socket_path: &str) {
+pub fn worker_entrypoint(socket_path: &str, node_version: Option<&str>) {
 	worker_event_loop("prepare", socket_path, |rt_handle, mut stream| async move {
+		let worker_pid = std::process::id();
+		let version_mismatch = if let Some(version) = node_version {
+			version != env!("SUBSTRATE_CLI_IMPL_VERSION")
```

Comment:
So there are two hypothetical cases that can trigger this:

1. Old node, new worker -> the old node should be shutdown
1. New node, old worker -> the worker should be shutdown (and probably any result discarded and the job re-tried)

Can 2 ever happen? If so, we should not shutdown the whole node in that case.

Reply:
2 can not happen as the new node doesn't know the old workers.

Reply:
Yes, good point. This should be documented.

Reply:
> Can 2 ever happen?

No, I don't how that's possible.

Reply:
Well, purely theoretically, one can downgrade the node in-place, then we'll get 2 :) But it's nothing different from the node upgrade scenario, we'll have to shutdown the node anyway. So I don't try to check if the worker version is greater or less, just "does not match" is enough.